stl/src/tzdb.cpp

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+// This must be as small as possible, because its contents are
+// injected into the msvcprt.lib and msvcprtd.lib import libraries.
+// Do not include or define anything else here.
+// In particular, basic_string must not be included here.
+
+#include <xtzdb.h>
+
+#include <Windows.h>
+
+#pragma comment(lib, "Advapi32")
+
+_EXTERN_C
+
+_RegistryLeapSecondInfo* __stdcall __std_tzdb_get_reg_leap_seconds(
+    const size_t prev_reg_ls_size, size_t* current_reg_ls_size) {
+    // On exit---
+    //    *current_reg_ls_size <= prev_reg_ls_size, *reg_ls_data == nullptr --> no new data
+    //    *current_reg_ls_size >  prev_reg_ls_size, *reg_ls_data != nullptr --> new data, successfully read
+    //    *current_reg_ls_size == 0,                *reg_ls_data != nullptr --> new data, failed reading
+    //    *current_reg_ls_size >  prev_reg_ls_size, *reg_ls_data == nullptr --> new data, failed allocation
+
+    constexpr auto reg_key_name    = TEXT("SYSTEM\\CurrentControlSet\\Control\\LeapSecondInformation");
+    constexpr auto reg_subkey_name = TEXT("LeapSeconds");
+    *current_reg_ls_size           = 0;
+    HKEY leap_sec_key              = 0;
+
+    LSTATUS status = RegOpenKeyEx(HKEY_LOCAL_MACHINE, reg_key_name, 0, KEY_READ, &leap_sec_key);

```

Comment:
We could specify the `W` version of Windows APIs. Same below.

```suggestion
    constexpr auto reg_key_name    = LR"(SYSTEM\CurrentControlSet\Control\LeapSecondInformation)";
    constexpr auto reg_subkey_name = L"LeapSeconds";
    *current_reg_ls_size           = 0;
    HKEY leap_sec_key              = 0;

    LSTATUS status = RegOpenKeyExW(HKEY_LOCAL_MACHINE, reg_key_name, 0, KEY_READ, &leap_sec_key);
```